src/common/dav/memops.c

```diff
@@ -531,7 +531,7 @@ operation_process_persistent_redo(struct operation_context *ctx)
 int
 operation_reserve(struct operation_context *ctx, size_t new_capacity)
 {
-	if (new_capacity > ctx->ulog_capacity) {
+	if ((ctx->type == LOG_TYPE_UNDO) && (new_capacity > ctx->ulog_capacity)) {
```

Comment:
I don't see why redo log doesn't need extending, could you explain it a bit?

Reply:
As mentioned in the previous comment, the redo ops do not update ctx->ulog. Instead they get appended to the log within ctx->pshadow_ops.  The log in pshadow_ops is extended using realloc within the function operation_add_typed_entry().